syntax = "proto3";

package llamaservice;

service LLamaService {
  rpc UploadImage (ImageRequest) returns (StatusResponse);
  rpc GetStatus (EmptyRequest) returns (StatusResponse);
  rpc Predict (PredictionRequest) returns (PredictionResponse);
  rpc ExportModel (ModelExportRequest) returns (StatusResponse);
  rpc CreateDataset (DatasetRequest) returns (StatusResponse);
  rpc ViewLogs (EmptyRequest) returns (LogResponse);
  rpc DownloadCheckpoint (EmptyRequest) returns (FileResponse);
}

message ImageRequest {
  string image_content = 1;
}

message EmptyRequest {}

message StatusResponse {
  string status = 1;
}

message PredictionRequest {
  string image_content = 1;
}

message PredictionResponse {
  string result = 1;
}

message ModelExportRequest {
  string format = 1;
}

message DatasetRequest {
  string dataset_name = 1;
}

message LogResponse {
  string logs = 1;
}

message FileResponse {
  bytes file_content = 1;
}
